homeassistant/components/statistics/sensor.py

```diff
@@ -231,6 +232,7 @@ def valid_boundary_configuration(config: dict[str, Any]) -> dict[str, Any]:
             vol.Coerce(int), vol.Range(min=1)
         ),
         vol.Optional(CONF_MAX_AGE): cv.time_period,
+        vol.Optional(CONF_PRESERVE_LAST_VAL, default=False): cv.boolean,
```

Comment:
Since configuring this is meaningless unless `max_age` is also configured, I think we should fail if the user sets this without also setting `max_age`.

Hence, please add a customer validator to `PLATFORM_SCHEMA` which checks this, you can check the `valid_state_characteristic_configuration` and `valid_boundary_configuration` for how to do this.

Reply:
Added validation & associated unit tests.